lib/bootloader_setup.pm

```diff
@@ -64,120 +78,154 @@ sub select_bootmenu_option {
         # and UEFI menues.
         send_key 'up';
     }
-    if (get_var('ZDUP') || get_var("ONLINE_MIGRATION")) {
-        send_key 'ret';    # boot from hard disk
+    if (get_var('ZDUP') || get_var('ONLINE_MIGRATION')) {
+        boot_local_disk;
         return 3;
     }
 
-    if (get_var("UPGRADE")) {
+    if (get_var('UPGRADE')) {
         # random magic numbers
-        send_key_until_needlematch('inst-onupgrade', 'down', 10, 5);
+        if (check_var('ARCH', 'ppc64le')) {
+            send_key_until_needlematch 'inst-onupgrade', 'up', 10, 5;
+        }
+        else {
+            send_key_until_needlematch 'inst-onupgrade', 'down', 10, 5;
+        }
     }
     else {
-        if (get_var("PROMO") || get_var('LIVETEST') || get_var('LIVE_INSTALLATION')) {
-            send_key_until_needlematch("boot-live-" . get_var("DESKTOP"), 'down', 10, 5);
+        if (get_var('PROMO') || get_var('LIVETEST') || get_var('LIVE_INSTALLATION')) {
+            send_key_until_needlematch 'boot-live-' . get_var('DESKTOP'), 'down', 10, 5;
         }
-        elsif (!get_var("JEOS")) {
-            send_key_until_needlematch('inst-oninstallation', 'down', 10, 5);
+        elsif (check_var('ARCH', 'ppc64le')) {
+            send_key_until_needlematch 'inst-oninstallation', 'up', 10, 5;
+        }
+        elsif (!get_var('JEOS')) {
+            send_key_until_needlematch 'inst-oninstallation', 'down', 10, 5;
```

Comment:
this is not if else case
leave this be, I will open poo to refactor whole boot mess, simplify, unify and everything possible